[BUG] Flaky upstream polars rolling test

**Describe the bug**
Example of the test failing: https://github.com/rapidsai/cudf/actions/runs/19144899197/job/54719782473

<details>

```
2025-11-06T18:06:51.1979722Z =================================== FAILURES ===================================
2025-11-06T18:06:51.1981485Z ____________________ test_rolling_agg_bad_input_types[str] _____________________
2025-11-06T18:06:51.1983686Z [gw5] linux -- Python 3.13.9 /pyenv/versions/3.13.9/bin/python
2025-11-06T18:06:51.1986189Z Traceback (most recent call last):
2025-11-06T18:06:51.1988299Z   File "/pyenv/versions/3.13.9/lib/python3.13/site-packages/_pytest/runner.py", line 341, in from_call
2025-11-06T18:06:51.1990363Z     result: TResult | None = func()
2025-11-06T18:06:51.1991737Z                              ~~~~^^
2025-11-06T18:06:51.1993756Z   File "/pyenv/versions/3.13.9/lib/python3.13/site-packages/_pytest/runner.py", line 242, in <lambda>
2025-11-06T18:06:51.1996359Z     lambda: runtest_hook(item=item, **kwds), when=when, reraise=reraise
2025-11-06T18:06:51.1998416Z             ~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^
2025-11-06T18:06:51.2000412Z   File "/pyenv/versions/3.13.9/lib/python3.13/site-packages/pluggy/_hooks.py", line 512, in __call__
2025-11-06T18:06:51.2003329Z     return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
2025-11-06T18:06:51.2007679Z            ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-11-06T18:06:51.2009226Z   File "/pyenv/versions/3.13.9/lib/python3.13/site-packages/pluggy/_manager.py", line 120, in _hookexec
2025-11-06T18:06:51.2010937Z     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
2025-11-06T18:06:51.2012325Z            ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-11-06T18:06:51.2013840Z   File "/pyenv/versions/3.13.9/lib/python3.13/site-packages/pluggy/_callers.py", line 167, in _multicall
2025-11-06T18:06:51.2015370Z     raise exception
2025-11-06T18:06:51.2016652Z   File "/pyenv/versions/3.13.9/lib/python3.13/site-packages/pluggy/_callers.py", line 139, in _multicall
2025-11-06T18:06:51.2018097Z     teardown.throw(exception)
2025-11-06T18:06:51.2018858Z     ~~~~~~~~~~~~~~^^^^^^^^^^^
2025-11-06T18:06:51.2020385Z   File "/pyenv/versions/3.13.9/lib/python3.13/site-packages/_pytest/threadexception.py", line 92, in pytest_runtest_call
2025-11-06T18:06:51.2022245Z     yield from thread_exception_runtest_hook()
2025-11-06T18:06:51.2024040Z   File "/pyenv/versions/3.13.9/lib/python3.13/site-packages/_pytest/threadexception.py", line 68, in thread_exception_runtest_hook
2025-11-06T18:06:51.2025844Z     yield
2025-11-06T18:06:51.2027063Z   File "/pyenv/versions/3.13.9/lib/python3.13/site-packages/pluggy/_callers.py", line 139, in _multicall
2025-11-06T18:06:51.2028511Z     teardown.throw(exception)
2025-11-06T18:06:51.2029285Z     ~~~~~~~~~~~~~~^^^^^^^^^^^
2025-11-06T18:06:51.2030852Z   File "/pyenv/versions/3.13.9/lib/python3.13/site-packages/_pytest/unraisableexception.py", line 95, in pytest_runtest_call
2025-11-06T18:06:51.2032798Z     yield from unraisable_exception_runtest_hook()
2025-11-06T18:06:51.2034699Z   File "/pyenv/versions/3.13.9/lib/python3.13/site-packages/_pytest/unraisableexception.py", line 70, in unraisable_exception_runtest_hook
2025-11-06T18:06:51.2037331Z     yield
2025-11-06T18:06:51.2038907Z   File "/pyenv/versions/3.13.9/lib/python3.13/site-packages/pluggy/_callers.py", line 139, in _multicall
2025-11-06T18:06:51.2040369Z     teardown.throw(exception)
2025-11-06T18:06:51.2041242Z     ~~~~~~~~~~~~~~^^^^^^^^^^^
2025-11-06T18:06:51.2042226Z   File "/pyenv/versions/3.13.9/lib/python3.13/site-packages/_pytest/logging.py", line 848, in pytest_runtest_call
2025-11-06T18:06:51.2043035Z     yield from self._runtest_for(item, "call")
2025-11-06T18:06:51.2043808Z   File "/pyenv/versions/3.13.9/lib/python3.13/site-packages/_pytest/logging.py", line 831, in _runtest_for
2025-11-06T18:06:51.2059714Z     yield
2025-11-06T18:06:51.2062045Z   File "/pyenv/versions/3.13.9/lib/python3.13/site-packages/pluggy/_callers.py", line 139, in _multicall
2025-11-06T18:06:51.2064233Z     teardown.throw(exception)
2025-11-06T18:06:51.2065631Z     ~~~~~~~~~~~~~~^^^^^^^^^^^
2025-11-06T18:06:51.2067765Z   File "/pyenv/versions/3.13.9/lib/python3.13/site-packages/_pytest/capture.py", line 879, in pytest_runtest_call
2025-11-06T18:06:51.2069978Z     return (yield)
2025-11-06T18:06:51.2070938Z             ^^^^^
2025-11-06T18:06:51.2073527Z   File "/pyenv/versions/3.13.9/lib/python3.13/site-packages/pluggy/_callers.py", line 139, in _multicall
2025-11-06T18:06:51.2075941Z     teardown.throw(exception)
2025-11-06T18:06:51.2077106Z     ~~~~~~~~~~~~~~^^^^^^^^^^^
2025-11-06T18:06:51.2079270Z   File "/pyenv/versions/3.13.9/lib/python3.13/site-packages/pluggy/_callers.py", line 53, in run_old_style_hookwrapper
2025-11-06T18:06:51.2081811Z     return result.get_result()
2025-11-06T18:06:51.2083003Z            ~~~~~~~~~~~~~~~~~^^
2025-11-06T18:06:51.2084959Z   File "/pyenv/versions/3.13.9/lib/python3.13/site-packages/pluggy/_result.py", line 103, in get_result
2025-11-06T18:06:51.2087610Z     raise exc.with_traceback(tb)
2025-11-06T18:06:51.2089827Z   File "/pyenv/versions/3.13.9/lib/python3.13/site-packages/pluggy/_callers.py", line 38, in run_old_style_hookwrapper
2025-11-06T18:06:51.2092317Z     res = yield
2025-11-06T18:06:51.2093237Z           ^^^^^
2025-11-06T18:06:51.2095502Z   File "/pyenv/versions/3.13.9/lib/python3.13/site-packages/pluggy/_callers.py", line 139, in _multicall
2025-11-06T18:06:51.2097675Z     teardown.throw(exception)
2025-11-06T18:06:51.2102937Z     ~~~~~~~~~~~~~~^^^^^^^^^^^
2025-11-06T18:06:51.2104885Z   File "/pyenv/versions/3.13.9/lib/python3.13/site-packages/_pytest/skipping.py", line 257, in pytest_runtest_call
2025-11-06T18:06:51.2107034Z     return (yield)
2025-11-06T18:06:51.2108577Z             ^^^^^
2025-11-06T18:06:51.2114124Z   File "/pyenv/versions/3.13.9/lib/python3.13/site-packages/pluggy/_callers.py", line 121, in _multicall
2025-11-06T18:06:51.2114896Z     res = hook_impl.function(*args)
2025-11-06T18:06:51.2115784Z   File "/pyenv/versions/3.13.9/lib/python3.13/site-packages/_pytest/runner.py", line 174, in pytest_runtest_call
2025-11-06T18:06:51.2116570Z     item.runtest()
2025-11-06T18:06:51.2116914Z     ~~~~~~~~~~~~^^
2025-11-06T18:06:51.2117558Z   File "/pyenv/versions/3.13.9/lib/python3.13/site-packages/_pytest/python.py", line 1627, in runtest
2025-11-06T18:06:51.2118341Z     self.ihook.pytest_pyfunc_call(pyfuncitem=self)
2025-11-06T18:06:51.2118856Z     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
2025-11-06T18:06:51.2119592Z   File "/pyenv/versions/3.13.9/lib/python3.13/site-packages/pluggy/_hooks.py", line 512, in __call__
2025-11-06T18:06:51.2120497Z     return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
2025-11-06T18:06:51.2121168Z            ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-11-06T18:06:51.2122108Z   File "/pyenv/versions/3.13.9/lib/python3.13/site-packages/pluggy/_manager.py", line 120, in _hookexec
2025-11-06T18:06:51.2143903Z     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
2025-11-06T18:06:51.2144801Z            ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-11-06T18:06:51.2146272Z   File "/pyenv/versions/3.13.9/lib/python3.13/site-packages/pluggy/_callers.py", line 167, in _multicall
2025-11-06T18:06:51.2147263Z     raise exception
2025-11-06T18:06:51.2148114Z   File "/pyenv/versions/3.13.9/lib/python3.13/site-packages/pluggy/_callers.py", line 121, in _multicall
2025-11-06T18:06:51.2149275Z     res = hook_impl.function(*args)
2025-11-06T18:06:51.2150271Z   File "/pyenv/versions/3.13.9/lib/python3.13/site-packages/_pytest/python.py", line 159, in pytest_pyfunc_call
2025-11-06T18:06:51.2151294Z     result = testfunction(**testargs)
2025-11-06T18:06:51.2152744Z   File "/__w/cudf/cudf/polars/py-polars/tests/unit/operations/test_rolling.py", line 140, in test_rolling_agg_bad_input_types
2025-11-06T18:06:51.2153891Z     with pytest.raises(TypeError):
2025-11-06T18:06:51.2154449Z          ~~~~~~~~~~~~~^^^^^^^^^^^
2025-11-06T18:06:51.2155551Z   File "/pyenv/versions/3.13.9/lib/python3.13/site-packages/_pytest/python_api.py", line 1011, in __exit__
2025-11-06T18:06:51.2156537Z     fail(self.message)
2025-11-06T18:06:51.2157015Z     ~~~~^^^^^^^^^^^^^^
2025-11-06T18:06:51.2157841Z   File "/pyenv/versions/3.13.9/lib/python3.13/site-packages/_pytest/outcomes.py", line 178, in fail
2025-11-06T18:06:51.2158820Z     raise Failed(msg=reason, pytrace=pytrace)
2025-11-06T18:06:51.2159726Z Failed: DID NOT RAISE <class 'TypeError'>
2025-11-06T18:06:51.2160452Z =========================== short test summary info ============================
2025-11-06T18:06:51.2161957Z FAILED py-polars/tests/unit/operations/test_rolling.py::test_rolling_agg_bad_input_types[str] - Failed: DID NOT RAISE <class 'TypeError'>
```

</details>


**Steps/Code to reproduce bug**
WIP

**Expected behavior**
pass

**Environment details**
polars 1.33.1
[cudf-polars-polars-tests / 13.0.1, 3.13, arm64, ubuntu24.04, l4, latest-driver, latest-deps](https://github.com/rapidsai/cudf/actions/runs/19144899197/job/54719782473#logs)

--- a/ci/test_cudf_polars_polars_tests.sh
+++ b/ci/test_cudf_polars_polars_tests.sh
@@ -39,31 +39,6 @@ git clone https://github.com/pola-rs/polars.git --branch "${TAG}" --depth 1
 rapids-logger "Install polars test requirements"
 # We don't need to pick up dependencies from polars-cloud, so we remove it.
 sed -i '/^polars-cloud$/d' polars/py-polars/requirements-dev.txt
-# Deltalake release 1.2.0 contains breaking changes for Polars.
-# Tracking issue: https://github.com/pola-rs/polars/issues/24872
-sed -i 's/^deltalake>=1.1.4/deltalake>=1.1.4,<1.2.0/' polars/py-polars/requirements-dev.txt
-
-# Regression in Pydantic 2.12.0 so we use a lower-bound of 2.12.1 to avoid that
-# and also to provide a new enough version to support Python 3.14
-sed -i 's/^pydantic>=2.0.0.*/pydantic>=2.12.1/' polars/py-polars/requirements-dev.txt
-
-# https://github.com/pola-rs/polars/issues/25772
-# Remove upper bound on aiosqlite once we support polars >1.36.1
-sed -i 's/^aiosqlite/aiosqlite>=0.21.0,<0.22.0/' polars/py-polars/requirements-dev.txt
-
-# Remove upper bound on pandas once we support 3.0.0+
-sed -i 's/^pandas$/pandas>=2.0,<2.4.0/' polars/py-polars/requirements-dev.txt
-
-# Remove upper bound on pandas-stubs once we support 3.0.0+
-sed -i 's/^pandas-stubs/pandas-stubs<3/' polars/py-polars/requirements-dev.txt
-
-# Pyparsing release 3.3.0 deprecates the enablePackrat method, which is used by the
-# version of pyiceberg that polars is currently pinned to. We can remove this skip
-# when we move to a newer version of polars using a pyiceberg where this issue is fixed
-# Currently pyparsing is only a transitive dependency via pyiceberg, so we just
-# tack on the constrained dependency at the end of the file since there is no
-# existing dependency to rewrite.
-echo "pyparsing>=3.0.0,<3.3.0" >> polars/py-polars/requirements-dev.txt
 
 # notes:
 #

--- a/python/cudf_polars/cudf_polars/testing/inject_gpu_engine.py
+++ b/python/cudf_polars/cudf_polars/testing/inject_gpu_engine.py
@@ -30,7 +30,6 @@ def pytest_addoption(parser: pytest.Parser) -> None:
         choices=("in-memory", "spmd"),
         help="Which GPU engine variant to inject globally.",
     )
-    # TODO: We never run with --inject-gpu-engine-blocksize in ci/run_cudf_polars_polars_tests.sh. Remove?
     group.addoption(
         "--inject-gpu-engine-blocksize",
         action="store",
@@ -284,7 +283,6 @@ TESTS_TO_SKIP: Mapping[str, str] = {
     "tests/benchmark/test_with_columns.py::test_with_columns_quadratic_19503": "Tests performance bug in CPU engine",
     # The test may segfault with the legacy streaming engine. We should
     # remove this skip when all polars tests use the new streaming engine.
-    "tests/unit/streaming/test_streaming_group_by.py::test_streaming_group_by_literal[1]": "May segfault w/the legacy streaming engine",
     # Fails in CI, but passes locally
     "tests/unit/streaming/test_streaming.py::test_streaming_streamable_functions": "RuntimeError: polars_python::sql::PySQLContext is unsendable, but is being dropped on another thread",
     # Remove when polars supports Pydantic V3
@@ -295,9 +293,6 @@ TESTS_TO_SKIP: Mapping[str, str] = {
     "tests/unit/series/test_describe.py::test_series_describe_float": "https://github.com/rapidsai/cudf/issues/19324",
     "tests/unit/series/test_describe.py::test_series_describe_int": "https://github.com/rapidsai/cudf/issues/19324",
     "tests/unit/streaming/test_streaming.py::test_streaming_apply": "https://github.com/pola-rs/polars/issues/22558",
-    # New iceberg release causes this test to fail. We can remove in the next polars version bump: https://github.com/rapidsai/cudf/pull/19912
-    "tests/unit/io/test_iceberg.py::test_fill_missing_fields_with_identity_partition_values[False]": "https://github.com/pola-rs/polars/pull/24456",
-    "tests/unit/operations/test_rolling.py::test_rolling_agg_bad_input_types[str]": "https://github.com/rapidsai/cudf/issues/20551",
     "tests/unit/operations/test_group_by_dynamic.py::test_group_by_dynamic_agg_bad_input_types[str]": "TODO: Need to investigate why this fails in CI but passes locally. We should fallback to CPU for group_by_dynamic",
     "tests/unit/expr/test_exprs.py::test_exp_log1p[Float16-Float16]": "Flaky test: Small floating-point precision differences in exp/log1p results",
     # TODO: Investigate why these tests fail in CI but pass locally.

